frontend/src/components/MailItem.tsx

```diff
@@ -68,10 +115,30 @@ export default function MailItem({
         handleConfirm={onConfirmRemove}
         handleCancel={onCancelRemove}
       />
+      <AbstractModal
+        title="View Profile"
+        isModalOpen={showProfile}
+        handleCancel={() => setShowProfile(false)}
+        isActionable={false}
+      >
+        <ProfileContent readonlyContent={profileContent} />
+      </AbstractModal>
       <Grid item container direction="row" xs={3} wrap="nowrap">
-        <AccountCircle fontSize="large" style={{ color: "#4b5e82" }} />
+        <IconButton onClick={() => setShowProfile(true)}>
+          <Avatar
+            src={profileContent.image ? profileContent.image : ""}
+            alt={profileContent.first_name}
+          />
+        </IconButton>
         <Typography variant="h6">{`${first_name} ${last_name}`}</Typography>
```

Comment:
I set the first and last name of one of the profiles as null and it still displayed the name on the dashboard but set it as blank on the view_profile abstract modal. Something to think about: do we even want people setting privacy on their first and last name?

Reply:
that's a good point, it would be hard to identify your penpals if multiple have not image and their names are hidden

Reply:
At a minimum, last name should be able to be hidden

Reply:
I agree with at least last name being hidden and first name being public at all times.